app/src/main/java/com/techyos/oneclickonemail/ui/widgets/WidgetProvider.java

```diff
@@ -26,7 +27,7 @@
 
     @Override
     public void onUpdate(Context context, AppWidgetManager appWidgetManager, int[] appWidgetIds) {
-        WidgetEntryDao dao = new WidgetEntryDao(new DbHelper(context));
+        WidgetEntryDao dao = ((App)context.getApplicationContext()).getWidgetEntryDao();
```

Comment:
I am not sure whether  `ApplicationContext.onCreate` is always called before `onUpdate`. Needs investigation.

:x: Do not yet merge.